src/app/DomainSettingsPage.qml

```diff
@@ -0,0 +1,390 @@
+                    }
+                    domainSettingsListView.ViewItems.selectedIndices = []
+                    selectMode = false
+                }
+            },
+            Action {
+                iconName: "edit"
+                visible: !selectMode
+                enabled: domainSettingsListView.count > 0
+                onTriggered: {
+                    selectMode = true
+                }
+            },
+            Action {
+                iconName: "add"
+                visible: !selectMode
+
+                onTriggered: {
+                    var promptDialog = PopupUtils.open(Qt.resolvedUrl("PromptDialog.qml"), domainSettingsPage);
+                    promptDialog.title = i18n.tr("Add domain")
+                    promptDialog.message = i18n.tr("Enter the name of the domain, e.g. m.example.com")
+                    promptDialog.accept.connect(function(text) {
+                        if (text !== "") {
+                            var domain = UrlUtils.extractHost(text)
+                            if (DomainSettingsModel.contains(domain)) {
+                                domainSettingsItem.setDomainAsCurrentItem(domain);
+                            }
+                            else {
+                                DomainSettingsModel.insertEntry(domain);
+                                reload(domain);
+                            }
+                        }
+                    });
+                }
+            }
+        ]
+
+        onBack: {
+            selectMode = false;
+            domainSettingsItem.done();
+        }
+
+        DomainSettingsSortedModel {
+            id: domainSettingsSortedModel
+            model: DomainSettingsModel
+            sortOrder: Qt.AscendingOrder
+        }
+
+        ListItem {
+            id: useragentsMenu
+            z: 3
+            // custom user agents deactivated for now
+            height: 0 //units.gu(6)
+            color: theme.palette.normal.background
+            ListItemLayout {
+                title.text: i18n.tr("Custom User Agents")
+                ProgressionSlot {}
+            }
+
+            onClicked: customUserAgentsViewLoader.active = true
+        }
+
+        ListView {
+            id: domainSettingsListView
+            anchors.topMargin: useragentsMenu.height
+            anchors.fill: parent
+            focus: true
+            model:  SortFilterModel {
+                model: domainSettingsSortedModel
+            }
+
+            ViewItems.selectMode: selectMode
+
+            delegate: ListItem {
+                id: item
+                readonly property bool isCurrentItem: item.ListView.isCurrentItem
+                readonly property string domain: model.domain
+                readonly property int userAgentId: model.userAgentId
+                height: isCurrentItem ? layout.height : units.gu(5)
+                color: isCurrentItem ? theme.palette.selected.base : theme.palette.normal.background
```

Comment:
Again, I think that `base` is much darker than you expect.

Reply:
yeah might need tweaking